libp2p/protocols/secure/noise.nim

```diff
@@ -286,38 +286,23 @@ proc sendHSMessage(sconn: Connection; buf: seq[byte]) {.async.} =
   await sconn.write(outbuf)
 
 proc packNoisePayload(payload: openarray[byte]): seq[byte] =
-  var
-    ns: uint32
-  if randomBytes(addr ns, 4) != 4:
-    raise newException(NoiseHandshakeError, "Failed to generate randomBytes")
-
-  let
-    noiselen = int((ns mod (NoiseSize - 2)) + 1)
-    plen = payload.len.uint16
-
-  var
-    noise = newSeq[byte](noiselen)
-  if randomBytes(noise) != noiselen:
-    raise newException(NoiseHandshakeError, "Failed to generate randomBytes")
-
-  result &= plen.toBytesBE
+  result &= payload.len.uint16.toBytesBE
   result &= payload
-  result &= noise
 
   if result.len > uint16.high.int:
     raise newException(NoiseOversizedPayloadError, "Trying to send an unsupported oversized payload over Noise")
 
-  trace "packed noise payload", inSize = payload.len, outSize = result.len
+  trace "packed noise payload", size = payload.len
 
 proc unpackNoisePayload(payload: var seq[byte]) =
   let
     besize = payload[0..1]
     size = uint16.fromBytesBE(besize).int
 
-  if size > (payload.len - 2):
+  if size != (payload.len - 2):
```

Comment:
@arnetheduck does this change makes sense to you? 
I mean I guess that's exactly correct now... but invalidates any possible (wrong) padding.

Reply:
doesn't the standard say what to do with padded packets? 

If it doesn't, I'd generally lean towards allowing padding but not producing it, meaning using the original condition here.

I'd also generally avoid mutating the passed-in `var` most likely, instead trying to use the data in-place without copying it - for example turning this into a template and using `toOpenArray` etc.

Reply:
Oh, and I'd also include a link to the standard and explain the situation in a comment.

Reply:
The spec does not say yeah, let's do like you said.

About the template, hmm what's the difference with `var` mutation? there should be no copy, but I will check again call-sites, indeed `toOpenArray` now is definitely an option too.